src/WebJobs.Extensions.DurableTask/DurableTaskExtension.cs

```diff
@@ -295,6 +296,30 @@ void IExtensionConfigProvider.Initialize(ExtensionConfigContext context)
             this.taskHubWorker.AddOrchestrationDispatcherMiddleware(this.EntityMiddleware);
             this.taskHubWorker.AddOrchestrationDispatcherMiddleware(this.OrchestrationMiddleware);
 
+            // Initialize the EventSourceListener
+            string containerName = this.nameResolver.Resolve("CONTAINER_NAME");
+            string azureWebsiteInstanceId = this.nameResolver.Resolve("WEBSITE_INSTANCE_ID");
+            string functionsLogsMountPath = this.nameResolver.Resolve("FUNCTIONS_LOGS_MOUNT_PATH");
+
+            bool inAppService = !string.IsNullOrEmpty(azureWebsiteInstanceId);
+            bool inLinuxDedicated = inAppService && !string.IsNullOrEmpty(functionsLogsMountPath);
+            bool inLinuxConsumption = !inAppService && !string.IsNullOrEmpty(containerName);
+
+            LinuxAppServiceLogger linuxLogger = null;
+            if (inLinuxDedicated)
+            {
+                linuxLogger = new LinuxAppServiceLogger(writeToConsole: false);
+            }
+            else if (inLinuxConsumption)
+            {
+                linuxLogger = new LinuxAppServiceLogger(writeToConsole: true);
+            }
+
+            if (linuxLogger != null)
+            {
+                this.eventSourceListener = new EventSourceListener(linuxLogger);
+            }

```

Comment:
Consider wrapping these up in a separate, private method since this method is already fairly long.